sbt-plugin/src/main/scala/org/scalajs/sbtplugin/Run.scala

```diff
@@ -0,0 +1,47 @@
+package org.scalajs.sbtplugin
+
+import org.scalajs.jsenv._
+
+import scala.concurrent._
+import scala.concurrent.duration.Duration
+import scala.concurrent.ExecutionContext.Implicits.global
+
+private[sbtplugin] object Run {
+  /** Starts and waits for a run on the given [[JSEnv]] interruptibly.
+   *
+   *  Interruption can be triggered by typing anything into stdin.
+   */
+  def runInterruptible(jsEnv: JSEnv, input: Input, config: RunConfig): Unit = {
```

Comment:
We talked about this:
https://github.com/scala-js/scala-js/issues/3033#issuecomment-326749049 :P

Note that in the current implementation we indeed do *not* redirect stdin.

The problem with "run to completion" is that it is not supported by webby JS envs (PhantomJS, Selenium).

> Node.js application that would actually read things from its standard input.

I really hope nobody builds this in SJS, just as a side note :P

Reply:
Ah, right. But the `NodeJSEnv` (not necessarily other JSEnvs) still does guarantee run-to-completion by default, doesn't it? Otherwise even our `helloworld/run` on the CI is not going to work.

Reply:
Yes. A JSEnv *may* provide run to completion by default.

Reply:
OK. It makes sense, now.